index.ts

```diff
@@ -0,0 +1,10 @@
+import { Platform } from 'react-native';
+import WebViewIOS from './src/WebView.ios';
+import WebViewAndroid from './src/WebView.android';
+
+const WebView = Platform.OS === 'android' ? WebViewAndroid : WebViewIOS;
```

Comment:
I might be missing something here, but won't this negate the automatic inclusion of only the correct platform for each bundle, and now always include both the Android and iOS code for both platforms? 🤔 

Would anything be lost by simply setting `"main": "src/WebView"`? 🤔

Also, and this is probably a much broader discussion, but I think that this is the first package in the React Native ecosystem to be published as pure TypeScript, instead of shipping the, by `tsc`, precompiled version. Is this something that we want to do? What are the pros/cons.

One con that I see is that it will probably mess up for people that are using a custom `.babelrc` that doesn't include TypeScript. I actually personally found it a problem that some React Native ships as flow code, which means that you cannot use a custom `.babelrc` that doesn't include flow.

The only pro I can think of is that it simplifies the publishing of this module a little bit, since we don't have to have a `"prepare"` script. Might be other pros that I'm missing though 🤔 

One approach would be to let `tsc` transpile the `src/` dir into `lib/`, and then have two small entry files:

**`index.js`**

```js
import WebView from './lib/WebView'

// We keep this for compatibility reasons
export { WebView }
export default WebView
```

**`index.ts`**

```ts
import WebViewIOS from './lib/WebView.ios'
import WebViewAndroid from './lib/WebView.android'

declare const WebView: WebViewIOS | WebViewAndroid

export { WebView }
export default WebView
```

The `lib/` folder would contain both `*.js` and `*.d.ts` files, which is the default output from `tsc`. We would just have to add `"prepare": "tsc"` to the package.json-scripts.